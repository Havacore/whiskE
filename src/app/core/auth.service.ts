import {Injectable} from '@angular/core';
import {Router} from '@angular/router';
import * as firebase from 'firebase/app';
import {AngularFireAuth} from 'angularfire2/auth';
import {AngularFirestore, AngularFirestoreDocument} from 'angularfire2/firestore';
import {Observable} from 'rxjs/Observable';
import 'rxjs/add/operator/switchMap';

interface User {
  uid: string;
  email: string;
  photoURL?: string;
  displayName?: string;
  favoriteColor?: string;
}

@Injectable()
export class AuthService {
  user: Observable<User>;
  private isLoggedIn: boolean;

  constructor(private afAuth: AngularFireAuth,
              private afs: AngularFirestore,
              private router: Router) {
    //// Get auth data, then get firestore user document || null
    this.user = this.afAuth.authState
      .switchMap(user => {
        if (user) {
          this.isLoggedIn = true;
          return this.afs.doc<User>(`users/${user.uid}`).valueChanges();
        } else {
          this.isLoggedIn = false;
          return Observable.of(null);
        }
      });
  }

  get currentUserObservable(): any {
  return this.afAuth.auth;
}

  set loggedIn(isLoggedIn) {
    this.isLoggedIn = isLoggedIn;
  }

  get loggedIn(): boolean {
    return this.isLoggedIn;
  }

  googleLogin() {
    const provider = new firebase.auth.GoogleAuthProvider();
    return this.oAuthLogin(provider);
  }

  private oAuthLogin(provider) {
    return this.afAuth.auth.signInWithPopup(provider)
      .then((credential) => {
        this.isLoggedIn = true;
        this.updateUserData(credential.user);
        this.router.navigate(['/dashboard']);
      });
  }

  private updateUserData(user) {
    // Sets user data to firestore on login
    const userRef: AngularFirestoreDocument<any> = this.afs.doc(`users/${user.uid}`);
    const data: User = {
      uid: user.uid,
      email: user.email,
      displayName: user.displayName,
      photoURL: user.photoURL
    };
    return userRef.set(data);
  }

  signOut() {
    this.afAuth.auth.signOut().then(() => {
      this.router.navigate(['/login']);
    });
  }
}
