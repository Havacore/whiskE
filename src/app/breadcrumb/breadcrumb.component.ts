import {Component, Input, OnInit} from '@angular/core';

export interface BreadCrumb {
  name: string;
  url: string;
}

@Component({
  selector: 'app-breadcrumb',
  template: `
    <div *ngFor="let crumb of crumbs">
      <div>{{crumb.name}}</div>
    </div>
  `,
  styleUrls: ['./breadcrumb.component.scss']
})
export class BreadcrumbComponent implements OnInit {
  @Input() crumbs: BreadCrumb[];

  constructor() {
  }

  ngOnInit() {
  }

}
