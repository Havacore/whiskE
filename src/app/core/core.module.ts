import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { AuthService } from './auth.service';
import {AuthGuard} from './auth.guard';
import {AngularFireAuthModule} from 'angularfire2/auth';
import {AngularFirestoreModule} from 'angularfire2/firestore';

@NgModule({
  imports: [
    CommonModule,
    AngularFireAuthModule,
    AngularFirestoreModule
  ],
  declarations: [],
  providers: [AuthService],
  exports: []
})
export class CoreModule { }
