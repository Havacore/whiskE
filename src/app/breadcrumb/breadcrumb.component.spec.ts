import { async, ComponentFixture, TestBed } from '@angular/core/testing';

import {BreadCrumb, BreadcrumbComponent} from './breadcrumb.component';

describe('BreadcrumbComponent', () => {
  let component: BreadcrumbComponent;
  let fixture: ComponentFixture<BreadcrumbComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ BreadcrumbComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(BreadcrumbComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('should display the one breadcrumb passed in', () => {
    component.crumbs = [{name: 'Dashboard', url: '/dashboard'}];
    fixture.detectChanges();
    const de = fixture.debugElement;
    expect(de.nativeElement.textContent).toContain('Dashboard');
    expect(fixture).toMatchSnapshot();
  });
});
