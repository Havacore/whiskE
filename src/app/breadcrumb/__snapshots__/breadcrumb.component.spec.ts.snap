// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadcrumbComponent should display the one breadcrumb passed in 1`] = `
<app-breadcrumb
  crumbs={[Function Array]}
>
  
      <div>
    
        
    <div>
      Dashboard
    </div>
    
      
  </div>
    
</app-breadcrumb>
`;
